libfwupdplugin/fu-input-stream.c

```diff
@@ -92,6 +95,247 @@ fu_input_stream_read_safe(GInputStream *stream,
 	return TRUE;
 }
 
+/**
+ * fu_input_stream_read_u8:
+ * @stream: a #GInputStream
+ * @offset: offset in bytes into @stream to copy from
+ * @value: (out) (nullable): the parsed value
```

Comment:
Why support this being nullable? It seems to me like it's pointless to do reads if they don't go anywhere ..

Reply:
In other code I've used it so we can say "if I was to read this address would it fail" -- but maybe for an input stream where we're trying to minimize reads it doesn't make sense. Shall I make all of them non-nullable?

Reply:
narrator: _he did that_